arkouda/util.py

```diff
@@ -282,13 +282,21 @@ def convert_if_categorical(values):
     return values
 
 
-def attach(name):
+def attach(name: str, dtype: str = "infer"):
     """
-    Attaches to a known element name without requiring to know if the element is a Strings object or pdarray
+    Attaches to a known element name. If a type is passed, the server will use that type
+    to pull the corresponding parts, otherwise the server will try to infer the type
     """
-    repMsg = generic_msg(cmd="attach", args=name)
-    dtype = repMsg.split()[2]
-    if dtype == "str":
-        return Strings.from_return_msg(repMsg)
+    repMsg = cast(str, generic_msg(cmd="genericAttach", args=f"{dtype}+{name}"))
+
+    if repMsg.split(";")[0] == "categorical":
+        return Categorical.from_return_msg(repMsg)
+    elif repMsg.split(";")[0] == "segarray":
```

Comment:
These will also need to be updated to split on `+`

Reply:
Updated to use `+` delimited string in the response message